node/src/blockchain/blockchain_interface/lower_level_interface.rs

```diff
@@ -1,21 +1,66 @@
 // Copyright (c) 2019, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
 
-use crate::blockchain::blockchain_interface::data_structures::errors::BlockchainResult;
+use actix::Recipient;
+use crate::blockchain::blockchain_interface::data_structures::errors::{BlockchainError, PayableTransactionError};
 use crate::sub_lib::wallet::Wallet;
-use ethereum_types::U64;
-use web3::types::U256;
+use ethereum_types::{H256, U64};
+use futures::Future;
+use web3::contract::Contract;
+use web3::transports::Http;
+use web3::types::{Address, Filter, Log, TransactionReceipt, U256};
+use masq_lib::blockchains::chains::Chain;
+use masq_lib::logger::Logger;
+use crate::accountant::db_access_objects::payable_dao::PayableAccount;
+use crate::blockchain::blockchain_bridge::PendingPayableFingerprintSeeds;
+use crate::blockchain::blockchain_interface::blockchain_interface_web3::lower_level_interface_web3::TransactionReceiptResult;
+use crate::blockchain::blockchain_interface::data_structures::ProcessedPayableFallible;
 
 pub trait LowBlockchainInt {
-    fn get_transaction_fee_balance(&self, wallet: &Wallet) -> ResultForBalance;
+    // TODO: GH-495 The data structures in this trait are not generic, will need associated_type_defaults to implement it.
+    // see issue #29661 <https://github.com/rust-lang/rust/issues/29661> for more information
 
-    fn get_service_fee_balance(&self, wallet: &Wallet) -> ResultForBalance;
+    fn get_transaction_fee_balance(
+        &self,
+        address: Address,
+    ) -> Box<dyn Future<Item = U256, Error = BlockchainError>>;
 
-    fn get_block_number(&self) -> LatestBlockNumber;
+    fn get_service_fee_balance(
+        &self,
+        address: Address,
+    ) -> Box<dyn Future<Item = U256, Error = BlockchainError>>;
 
-    fn get_transaction_id(&self, wallet: &Wallet) -> ResultForNonce;
-}
+    fn get_gas_price(&self) -> Box<dyn Future<Item = U256, Error = BlockchainError>>;
+
+    fn get_block_number(&self) -> Box<dyn Future<Item = U64, Error = BlockchainError>>;
+
+    fn get_transaction_id(
+        &self,
+        address: Address,
+    ) -> Box<dyn Future<Item = U256, Error = BlockchainError>>;
+
+    fn get_transaction_receipt(
+        &self,
+        hash: H256,
+    ) -> Box<dyn Future<Item = Option<TransactionReceipt>, Error = BlockchainError>>;
 
-pub type ResultForBalance = BlockchainResult<web3::types::U256>;
-pub type ResultForBothBalances = BlockchainResult<(web3::types::U256, web3::types::U256)>;
-pub type ResultForNonce = BlockchainResult<U256>;
-pub type LatestBlockNumber = BlockchainResult<U64>;
+    fn get_transaction_receipt_batch(
+        &self,
+        hash_vec: Vec<H256>,
+    ) -> Box<dyn Future<Item = Vec<TransactionReceiptResult>, Error = BlockchainError>>;
```

Comment:
It's dubious that we'll come out of this with two methods, and not less, both devoted to the transaction receipts. My guts feeling says you most likely need to decide only for one and let the other one go. 

Otherwise, good job here regarding the LowBlockchainInt's new face. I like it. I also wish a native speaker could invent a smoother name, because I always feel tickled when I read or hear it. It probably deserves better wording than just the prefix "lower-level-". I dunno. You tell me :angel: